### PR TITLE
Improved parsing for data URL scheme.

### PR DESCRIPTION
**Pull request by anikiki**
Task/Issue URL: https://app.asana.com/0/1200905986587319/1203594326324933/f

### Description
Updated the data URI parsing.

### Steps to test this PR

- [x] Install from this branch and download a blob. I.e https://dopiaza.org/tools/datauri/examples/index.php (just smoke test example, we have tests covering this)

### CODE REVIEW

**Review by malmstein — COMMENTED** [review]
Shouldn't we have a test against this specific case?

**Comment by anikiki** [comment]
> Shouldn't we have a test against this specific case?

We have tests. This is just removing the pattern matching on large data.

File: downloads/downloads-impl/src/main/java/com/duckduckgo/downloads/impl/DataUriParser.kt [hunk 1]
```diff
@@ -26,15 +26,25 @@ import okio.ByteString.Companion.decodeBase64
...
-        val result = MIME_TYPE_REGEX.find(url)
+        val offset = url.indexOf(',')
+        if (offset == -1) {
+            return ParseResult.Invalid
+        }
+
+        val data = url.substring(offset + 1)
+        if (data.isEmpty()) {
+            return ParseResult.Invalid
+        }
```
Comment (joshliebe) [inline downloads/downloads-impl/src/main/java/com/duckduckgo/downloads/impl/DataUriParser.kt:37]: I think the unit test that's already there catches the first `ParseResult.Invalid` case, should we not add second test for this case?

**Review by joshliebe — APPROVED** [review]
Looks good and works as expected! Just added one comment.

**Pull request merged**